zklayer-whitepaper.tex

```diff
@@ -184,39 +184,39 @@ \subsubsection{Blockchain}
 
 A naïve solution would involve the model developer sharing the model with the customers, who would then run the model locally to ensure its integrity. However, this approach is not practical due to the large size of models and the limited computational resources available to customers. As an alternative, Ghodsi et al.~\cite{Ghodsi2017SafetyNetsVE} proposed, for the first time, the use of ZKP to design a solution where the model developer, or a third party like a cloud service, runs the model but generates a proof using ZKP for the customer to ensure that the genuine model has been executed.
 
-While Ghodsi's solution offers benefits, it necessitates the sharing of the model with customers. However, in certain scenarios, model developers may be reluctant to share their models due to concerns about protecting intellectual property. Consequently, in addition to ensuring the integrity of the model, preserving the privacy of the model becomes another significant concern. Thus, the overarching question arises: How can model developers assure customers that the model has been genuinely executed (ensuring the integrity of inference) without divulging any information about the model itself (preserving model privacy)? To address this new complicated problem, Lee et al.~\cite{Lee2020vCNNVC} proposed a ZKP-based solution which provide both concerns. Subsequently, other researchers have proposed more efficient ZKP-based solution to address these two concerns~\cite{Liu2021zkCNNZK},~\cite{Feng2021ZENAO},~\cite{Ju2021EfficientSP}.
+While Ghodsi's solution offers benefits, it necessitates the sharing of the model with customers. However, in certain scenarios, model developers may be reluctant to share their models due to concerns about protecting intellectual property. Consequently, in addition to ensuring the integrity of the model, preserving the privacy of the model becomes another significant concern. Thus, the overarching question arises: How can model developers assure customers that the model has been genuinely executed (ensuring the integrity of inference) without divulging any information about the model itself (preserving model privacy)? To address this new complicated problem, Lee et al.~\cite{Lee2020vCNNVC} proposed a ZKP-based solution that aims to addresse both concerns. Subsequently, other researchers have proposed a more efficient ZKP-based solution to address these two concerns~\cite{Liu2021zkCNNZK},~\cite{Feng2021ZENAO},~\cite{Ju2021EfficientSP}.
```

Comment:
```suggestion
While Ghodsi's solution offers benefits, it necessitates the sharing of the model with customers. However, in certain scenarios, model developers may be reluctant to share their models due to concerns about protecting intellectual property. Consequently, in addition to ensuring the integrity of the model, preserving the privacy of the model becomes another significant concern. Thus, the overarching question arises: How can model developers assure customers that the model has been genuinely executed (ensuring the integrity of inference) without divulging any information about the model itself (preserving model privacy)? To address this new complicated problem, Lee et al.~\cite{Lee2020vCNNVC} proposed a ZKP-based solution that aims to address both concerns. Subsequently, other researchers have proposed a more efficient ZKP-based solution to address these two concerns~\cite{Liu2021zkCNNZK},~\cite{Feng2021ZENAO},~\cite{Ju2021EfficientSP}.
```

Reply:
Although I don't particularly like the wording of

> To address this new complicated problem, Lee et al... proposed a ... solution that aims to address both concerns.

"To address ... solution that aims to address"

Reply:
fixed